cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/NamespaceSourceControlMetadataStore.java

```diff
@@ -0,0 +1,210 @@
+   * non-null when the application is pulled from the remote repository and deployed in the
+   * namespace.
+   *
+   * @param appRef            {@link ApplicationReference} for which the source control metadata is being
+   *                          set.
+   * @param sourceControlMeta The {@link SourceControlMeta} to be set. Can be {@code null} if
+   *                          application is just deployed.
+   * @throws IOException If failed to write the data.
+   */
+  public void write(ApplicationReference appRef,
+      @Nullable SourceControlMeta sourceControlMeta)
+      throws IOException {
+    // In the Namespace Pipelines page, the sync status (SYNCED or UNSYNCED)
+    // and last modified of all the applications deployed in the namespace needs to be shown.
+    // If source control information is not added when the app is deployed, the data will
+    // split into two tables.
+    // JOIN operation is not currently supported yet. The filtering (eg,
+    // filter on UNSYNCED sync status) , sorting, searching , pagination becomes difficult.
+    // Instead of doing filtering, searching, sorting in memory, it will happen at
+    // database level.
+    StructuredTable scmTable = getNamespaceSourceControlMetadataTable();
+    scmTable.upsert(getAllFields(appRef, sourceControlMeta));
+  }
+
+  /**
+   * Deletes the source control metadata associated with the specified application ID from the
+   * namespace.
+   *
+   * @param appRef {@link ApplicationReference} whose source control metadata is to be deleted.
+   * @throws IOException if it failed to read or delete the metadata
+   */
+  public void delete(ApplicationReference appRef) throws IOException {
+    getNamespaceSourceControlMetadataTable().delete(getPrimaryKey(appRef));
+  }
+
+  /**
+   * Deletes all rows of source control metadata within the specified namespace.
+   *
+   * @param namespace The namespace for which all source control metadata rows are to be deleted.
+   * @throws IOException if it failed to read or delete the metadata.
+   */
+  public void deleteAll(String namespace) throws IOException {
+    getNamespaceSourceControlMetadataTable().deleteAll(getNamespaceRange(namespace));
+  }
+
+  private Collection<Field<?>> getAllFields(ApplicationReference appRef,
+      SourceControlMeta scmMeta) throws IOException {
+    List<Field<?>> fields = getPrimaryKey(appRef);
+    fields.add(Fields.stringField(
+        StoreDefinition.NamespaceSourceControlMetadataStore.SPECIFICATION_HASH_FIELD,
+        scmMeta == null ? "" : scmMeta.getFileHash()));
+    fields.add(
+        Fields.stringField(StoreDefinition.NamespaceSourceControlMetadataStore.COMMIT_ID_FIELD,
+            scmMeta == null ? "" : scmMeta.getCommitId()));
+    // Whenever an app is deployed, the expected behavior is that the last modified field will be
+    // retained and not reset.
+    Long lastModified = 0L;
```

Comment:
@samdgupi 
By default when an app is deployed and not already present in the new table, the value for last modified is kept as 0 and not null. I did this cause rows can be sorted on this field. Just wanted to let you know explicitly. Let me know if any changes are needed